src/datasets/features.py

```diff
@@ -549,53 +549,42 @@ class ClassLabel:
             classes, one per line.
     """
 
-    num_classes: Optional[int] = None
-    names: Optional[List[str]] = None
+    num_classes: int = None
+    names: List[str] = None
     names_file: Optional[str] = None
     id: Optional[str] = None
     # Automatically constructed
     dtype: ClassVar[str] = "int64"
     pa_type: ClassVar[Any] = pa.int64()
-    _str2int: ClassVar[Optional[Dict[str, int]]] = None
-    _int2str: ClassVar[Optional[Dict[int, int]]] = None
+    _str2int: ClassVar[Dict[str, int]] = None
+    _int2str: ClassVar[Dict[int, int]] = None
     _type: str = field(default="ClassLabel", init=False, repr=False)
 
     def __post_init__(self):
-        # The label is explicitly set as undefined (no label defined)
-        if not sum(bool(a) for a in (self.num_classes, self.names, self.names_file)):
-            return
-
-        # if sum(bool(a) for a in (self.num_classes, self.names, self.names_file)) != 1:
-        #     raise ValueError("Only a single argument of ClassLabel() should be provided.")
-
-        if self.num_classes is None:
-            if self.names is None:
+        if self.names_file is not None and self.names is not None:
+            raise ValueError("Please provide either names or names_file but not both.")
+        # Set self.names
+        if self.names is None:
+            if self.names_file is not None:
                 self.names = self._load_names_from_file(self.names_file)
-        else:
-            if self.names is None:
+            elif self.num_classes is not None:
                 self.names = [str(i) for i in range(self.num_classes)]
-            elif len(self.names) != self.num_classes:
-                raise ValueError(
-                    "ClassLabel number of names do not match the defined num_classes. "
-                    "Got {} names VS {} num_classes".format(len(self.names), self.num_classes)
-                )
-
+            else:
```

Comment:
wonder if it might be easier to do something like `assert any([num_classes, names, names_file])` at the top of the method to enforce the check that one of the three should exist.  (I guess it would have to be refactored a bit to ensure that it's exactly one of the three(?))

Reply:
I think it's fine this way
Plus it allows to have both names and num_classes as long as num_classes == len(names)